src/rest_api/classes/person/widgets/lineage.clj

```diff
@@ -0,0 +1,383 @@
+(ns rest-api.classes.person.widgets.lineage
+  (:require
+    [cheshire.core :as json]
+    [clojure.string :as str]
+    [datomic.api :as d]
+    [rest-api.db.main :refer [datomic-conn]]
+    [pseudoace.utils :as pace-utils]
+    [rest-api.formatters.date :as date]))
+
+(defn- generate-map [oid]
+  (let [other-person-id (get oid 0)
+        other-person-name (get oid 1)
+        json-data (get oid 2)
+        roles (filter #(re-find #"role" (str %)) (keys json-data))
+        level
+        (for [role roles
+              :let
+              [level (-> role
+                         (str/replace #":role/" "")
+                         (str/replace #"-" " ")
+                         (str/capitalize))]] level)
+        duration
+        (for [role roles
+              :let [rfrom (keyword
+                            (str
+                              (-> role (str/replace #"/" ".") (str/replace #":" "")) "/from"))
+                    rto (keyword
+                          (str
+                            (-> role (str/replace #"/" ".") (str/replace #":" "")) "/to"))
+                    from (if (contains? (first (json-data role)) rfrom)
+                           (date/format-date5 ((first (json-data role)) rfrom)))
+                    to (if (contains? (first (json-data role)) rto) 
+                         (date/format-date5 ((first (json-data role)) rto)))
+                    duration (str from " - " to)]] duration)]
+    (pace-utils/vmap
+      :level level
+      :duration duration
+      :name
+      (pace-utils/vmap
+        :label other-person-name
+        :id other-person-id
+        :taxonomy "all"
+        :class "person"))))
+
+(defn supervised-by [person]
+  (let [db (d/entity-db person)
+        data (->> (d/q '[:find ?other-person-id ?other-person-name (pull ?person-supervised-by-ent [*])
+                         :in $ ?person
+                         :where
+                         [?person :person/supervised-by ?person-supervised-by-ent]
+                         [?person-supervised-by-ent :person.supervised-by/person ?other-person-ent]
+                         [?other-person-ent :person/id ?other-person-id]
+                         [?other-person-ent :person/standard-name ?other-person-name]]
+                       db (:db/id person))
+                  (map (fn [oid]
+                         (generate-map oid)))
+                  (seq))]
+    {:data (if data data)
```

Comment:
The `if` here is redundant (since `seq` has been applied in the expression which binds `data` above).
i.e `(seq empty-collection)` returns `nil`

Reply:
Cool, that's good to know.  Fixed.